util.go

```diff
@@ -12,14 +12,44 @@ import (
 
 type contextKey string
 
-// SnowflakeRequestIDKey is optional context key to specify request id
-const SnowflakeRequestIDKey contextKey = "SNOWFLAKE_REQUEST_ID"
+const (
+	// MultiStatementCount controls the number of queries to execute in a single API call
+	MultiStatementCount paramKey = "MULTI_STATEMENT_COUNT"
+	// AsyncMode tells the server to not block the request on executing the entire query
+	asyncMode paramKey = "ASYNC_MODE_QUERY"
```

Comment:
Should asyncMode be public so the users of gosnowflake could set this param?

Reply:
I think they should all be private with getter/setter functions to access them. Not sure if that reasoning works with the stuff that @sfc-gh-dwu added though.